src/main/java/com/johnstarich/moviematcher/models/User.java

```diff
@@ -242,11 +242,69 @@ public User removeFriends(Collection<User> oldFriends) {
 		return new User(_id, email, first_name, last_name, newFriends, groups, password).updateFriends();
 	}
 
-	/* need to build Group class in order to implement */
-	public void addGroup() {}
-	public void addGroups() {}
-	public void removeGroup() {}
-	public void removeGroups() {}
+	public User addGroup(Group group) {
+		ArrayList<Group> groups = new ArrayList<>(this.groups);
+		groups.add(group);
+		return new User(_id, email, first_name, last_name, friends, groups, password).updateGroups();
+	}
+
+	public User addGroups(Collection<Group> groups) {
+		ArrayList<Group> newGroups = new ArrayList<>(this.groups);
+		newGroups.addAll(groups);
+		return new User(_id, email, first_name, last_name, friends, newGroups, password).updateGroups();
+	}
+
+	public User removeGroup(Group group) {
+		ArrayList<Group> groups = new ArrayList<>(this.groups);
+		groups.remove(group);
+		return new User(_id, email, first_name, last_name, friends, groups, password).updateGroups();
+	}
+
+	public User removeGroups(Collection<Group> groups) {
+		ArrayList<Group> newGroups = new ArrayList<>(this.groups);
+		newGroups.removeAll(groups);
+		return new User(_id, email, first_name, last_name, friends, newGroups, password).updateGroups();
```

Comment:
Remove the updateGroups calls from modifying methods. We'll call them ourselves when we want to make DB calls